tests/phpunit/tests/strings-in-block/test-html.php

```diff
@@ -409,6 +409,43 @@ public function it_should_update_innerContent_containing_a_slash() {
 		$this->assertEquals( '<div>' . $translated_block_inner_HTML . '</div>', $updated_block->innerHTML );
 	}
 
+	/**
+	 * @test
+	 * @group wpmlcore-6643
+	 */
+	public function it_should_update_and_NOT_escape_invalid_href() {
+		$block_name = 'core/some_block';
+
+		$config_option = \Mockery::mock( 'WPML_Gutenberg_Config_Option' );
+		$config_option->shouldReceive( 'get' )
+		              ->andReturn( [ $block_name => [ 'xpath' => [ '//div' ] ] ] );
+
+		$strings_in_block = new HTML( $config_option );
+
+		$target_lang                 = 'de';
+		$original_block_inner_HTML   = '<a href="[some-shortcode]">Click me</a>';
+		$translated_block_inner_HTML = '<a href="[some-shortcode]">Click me translated</a>';
+
+
+		$strings = array(
+			md5( $block_name . $original_block_inner_HTML ) => array(
+				$target_lang => array(
+					'value'  => $translated_block_inner_HTML,
+					'status' => (string) ICL_TM_COMPLETE,
+				)
+			)
+		);
+
+		$block               = \Mockery::mock( 'WP_Block_Parser_Block' );
+		$block->blockName    = $block_name;
+		$block->innerHTML    = '<div>' . $original_block_inner_HTML . '</div>';
+		$block->innerContent = [ $block->innerHTML ];
+
+		$updated_block = $strings_in_block->update( $block, $strings, $target_lang );
+
+		$this->assertEquals( '<div>' . $translated_block_inner_HTML . '</div>', $updated_block->innerHTML );
+	}
```

Comment:
@strategio i suggest adding a similar test where you assert that the returned string is **not the same** as the escaped one.
It might seem silly to do that, but I believe it would make clearer what issue we are solving by using `saveXML` instead of `saveHTML`.

Something like that:

```php
$expected = ’<a class="tb-button__link" href="%5Btb-dynamic%20provider='__current_post'%20post='current'%20source='toolset_custom_field%7Call-fields'%20field='url'%20force-string='first'%20%5D"><span class="tb-button__content">Pincha aquí</span></a>’n

$this->assertNotSame( $expected, $updated_block->innerHTML );
```

That would work more as a documentation test than an actual test.

Reply:
Done @andreasciamanna, indeed it's useful for documentation.